src/launcher/features/initialisation/initialiseLauncherState.ts

```diff
@@ -78,8 +78,22 @@ const loadTokenInformation = (): AppThunk => async (dispatch, getState) => {
     if (!shouldCheckForUpdatesAtStartup) return;
 
     try {
-        const token = await artifactoryToken.getTokenInformation();
-        if (token != null) dispatch(setArtifactoryTokenInformation(token));
+        const informationResult = await artifactoryToken.getTokenInformation();
+
+        if (informationResult.type === 'Success')
+            dispatch(
+                setArtifactoryTokenInformation(informationResult.information)
+            );
+
+        if (informationResult.type === 'Encryption not available') {
+            dispatch(
+                ErrorDialogActions.showDialog(
+                    'You must grant nRF Connect for Desktop permission to access your Keychain. ' +
+                        'Otherwise restricted sources will not be updated and ' +
+                        'trying to install apps from them will fail.'
```

Comment:
1) I'd consider changing the text to "nRF Connect for Desktop needs permission to access your Keychain. Without this permission, restricted sources won't be updated, and attempts to install apps from them will fail." or similar, so it has not "must" in it and sounds nicer :)

2) but modal lacks clear action for the user like triggering keychain again... is that possible to trigger again? on a mac cmd+R doesn't trigger, has to stop process and start again (dev build). We need to add clear next step for the user or explain what to do..

Reply:
Can we have "Close" and "Restart" buttons? Maybe with https://www.electronjs.org/docs/latest/api/app#apprelaunchoptions (might be not worth an effort)

Reply:
Given that this is uses the general purpose error dialog, there is no way with very little effort to add those buttons. 

And I don't know whether it would help. Since I cannot reproduce this anymore, I also cannot test this. Could you enter `const {app} = require('@electron/remote'); app.relaunch(); app.quit()` in the developer console and check whether this leads to the macOS asking for that permission again?